### DISCUSSION TITLE
I want to use the ADetailer API

### DISCUSSION THREAD

**Discussion by daikimasaokasiki** [Bing-su/adetailer#372] [Q&A] [answered]
The following parameters were written so that the API of ADetailer, an extension of Stable Diffusion, can be used with i2i.

```
      params = {
        ...sdParameters,
        init_images: [image.toString("base64")],
        denoising_strength: denoisingStrength,
        width: adjustWidth,
        height: adjustHeight,
        alwayson_scripts: {
          ADetailer: {
            args: [
              {
                ad_model: "face_yolov8n.pt",
                ad_mask_max_ratio: 0.03,
                ad_denoising_strength: denoisingStrength,
              }
            ]
          }
        }
      }
```
However, when we ran i2i with these parameters, ADetailer did not work. I would like to know if there is anything wrong with the parameters I have written regarding ADetailer, and if so, please correct it and let me know. The denoisingStrength is an array of 4 keys and values, all 4 set to different DS values. I would also like to know how to write parameters for ADetailer.

**Comment by daikimasaokasiki** [comment, marked as answer]
Resolved.All I had to do was add a double quotation.

**Reply by vvanhee** [reply]
Where did you add the double quotation? Thanks.

**Reply by daikimasaokasiki** [reply]
```
        "alwayson_scripts": {
          "ADetailer": {
            "args": [
              {
                "ad_model": "face_yolov8n.pt",
                "ad_mask_max_ratio": 0.03,
                "ad_denoising_strength": denoisingStrength,
              }
            ]
          }
        }
      }
```
I have added double quotation marks like this.

**Comment by vvanhee** [comment]
Thanks!
